### PR TITLE
[feat] Transcription Agent example!

### PR DESCRIPTION
**Pull request by om-raheja**
Transcription Agent
Description: Can listen to the microphone and execute tasks with the output. On request of @mannanshukla 
Dependencies: RealtimeSTT

Pretty self explanatory. Tested and works, if any issues come up please don't hesitate to @ me.

### CODE REVIEW

File: pyopenagi/tools/transcriber/transcriber.py
Comment (dongyuanjushi) [inline pyopenagi/tools/transcriber/transcriber.py:1]: The tool class for the transcriber is shown as CurrencyConverter, which is incorrect

**Review by dongyuanjushi — CHANGES_REQUESTED** [review]
Could you make the changes according to the above comments?

**Comment by om-raheja** [comment]
Hey Kai, I renamed the class to Transcriber before submitting the PR and I can't find where it says CurrencyConverter, neither locally nor in my diff nor on GitHub. I made some other needed adjustments to ensure code quality this morning, and I double checked all my class names. If there is anything else, please let me know. Sorry for the inconvenience.

**Comment by mannanshukla** [comment]
The class you used is called CurrencyConverter, just rename it.Sent from my iPhoneOn Aug 21, 2024, at 10:20 AM, om-raheja ***@***.***> wrote:﻿
Hey Kai, I renamed the class to Transcriber before submitting the PR and I can't find where it says CurrencyConverter, neither locally nor in my diff nor on GitHub. I made some other needed adjustments to ensure code quality this morning, and I double checked all my class names. If there is anything else, please let me know. Sorry for the inconvenience.

**Review by BRama10 — CHANGES_REQUESTED** [review]
In `interact.py`, have `stderr` still print to the console (or at worst, to a logfile), so that in the case of an error due to dependencies (which has happened many times before).

On second thought, also have `stdout` go to a logfile as well.

In `server.py` can you set up default arguments, and only in case of an error (such as the one you were getting), do we fallback onto parser args (since it is a hassle to pass args).

**Comment by om-raheja** [comment]
Balaji, will do the changes for interact.py. The way the parser is setup is that if no args are passed, the defaults you had before are automatically applied. Check the arg parser code.I can't find where the CurrencyConverter class is. Like I said before, its not in my diffs, my local files, or in GitHub. If you can show me where it is, I'll gladly change it.On Aug 21, 2024, at 11:43, BRama10 ***@***.***> wrote:﻿
@BRama10 requested changes on this pull request.

In interact.py, have stderr still print to the console (or at worst, to a logfile), so that in the case of an error due to dependencies (which has happened many times before).
On second thought, also have stdout go to a logfile as well.
In server.py can you set up default arguments, and only in case of an error (such as the one you were getting), do we fallback onto parser args (since it is a hassle to pass args).

**Comment by BRama10** [comment]
> Balaji, will do the changes for interact.py. The way the parser is setup is that if no args are passed, the defaults you had before are automatically applied. Check the arg parser code.I can't find where the CurrencyConverter class is. Like I said before, its not in my diffs, my local files, or in GitHub. If you can show me where it is, I'll gladly change it.On Aug 21, 2024, at 11:43, BRama10 ***@***.***> wrote:﻿ @BRama10 requested changes on this pull request. In interact.py, have stderr still print to the console (or at worst, to a logfile), so that in the case of an error due to dependencies (which has happened many times before). On second thought, also have stdout go to a logfile as well. In server.py can you set up default arguments, and only in case of an error (such as the one you were getting), do we fallback onto parser args (since it is a hassle to pass args). —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

I don't think CurrencyConverter ever was a problem (for some reason, your 2nd commit has CurrencyConverter, but your 3rd commit renames it to Transcriptor, so I assume you committed halfway through refactoring the class), so you can safely ignore that!

**Review by dongyuanjushi — APPROVED** [review]

**Review by BRama10 — APPROVED** [review]

**Pull request merged**